python/Ganga/GPIDev/Lib/Registry/RegistrySlice.py

```diff
@@ -304,7 +301,8 @@ def copy(self, keep_going):
         return this_slice
 
     def __contains__(self, j):
-        return j.id in self.objects.keys()
+        allObjK = self.objects.keys()
+        return j.id in allObjK
```

Comment:
Yes, the thing `Y` in a `for X in Y:` will be assigned to a temporary internal variable and have `iter()` called on it which then has `next()` called on it until it raises a `StopIteration` exception. In this case, `keys()` returns a `list` which which, when passed to `iter()` returns a `listiterator` object. Something like:
```python
for X in Y:
    ... something using X ...

→

_Y = iter(Y)
while True:
    try:
        X = next(_Y)
        ... something using X ...
    except StopIteration:
        break
```
so `Y` is only accessed once.